### PR TITLE
Python: Remove sync function invoke. Update other code to use async invoke.

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

Completes #4794. 

Most of SK Python is written with async code; however, there were some code paths that masked the underlying async code and made it look like a sync function invoke. In reality, asyncio spun up event loops and attempted to manage them properly. This becomes tricky when running a Jupyter notebook as they have their own event loop.

### Description

This PR removes the ability to call a kernel function synchronously. Since the majority of the code base is written with async methods, and to match dotnet, which is async, we're removing the sync function invoke and its dependent code. Some Planners were also run synchronously -- these were updated to run asynchronously. Kernel examples, notebooks, docs, and tests were updated accordingly. 

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Review by juliomenendez — APPROVED** [review]
LGTM

**Review by juliomenendez — APPROVED** [review]

File: python/semantic_kernel/orchestration/kernel_function.py [hunk 1]
```diff
@@ -307,7 +307,7 @@ def describe(self) -> FunctionView:
             parameters=self._parameters,
         )
 
-    def __call__(
+    async def __call__(
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/orchestration/kernel_function.py:310]: I'm not sure if __call__ is allowed to be overwritten as async func...

Comment (eavanvalkenburg) [reply python/semantic_kernel/orchestration/kernel_function.py:310]: alternative would be to just delete it (there is also no equivalent in dotnet)

**Review by eavanvalkenburg — CHANGES_REQUESTED** [review]
just a question on whether async __call__ is allowed, otherwise LGTM

File: python/semantic_kernel/orchestration/kernel_function.py (hunk 1 above)
Comment (moonbox3) [reply python/semantic_kernel/orchestration/kernel_function.py:310]: It works well to override as an async call. One can then `await some_func()`. The code has been updated from the previous sync override to async and notebooks, examples, and tests pass.

**Review by alexchaomander — APPROVED** [review]
LGTM!

**Review by eavanvalkenburg — APPROVED** [review]
Lgtm

**Pull request merged**